Jenkins plugin ImageStream SCM extended tests

csrwng opened the pull request:
Adds extended tests to validate the functionality of the Jenkins plugin imagestream SCM and corresponding DSL step.

csrwng commented:
Depends on https://github.com/openshift/origin/pull/11252

csrwng commented:
[testextended][extended:core(imagestream SCM)]

csrwng commented:
I've now rebased on the latest master
@gabemontero @jupierce ptal

jupierce commented on test/extended/image_ecosystem/plugin.go:
```diff
@@ -770,6 +771,65 @@ var _ = g.Describe("[image_ecosystem][jenkins][Slow] openshift pipeline plugin",
+			// Now wait after the image has been tagged for the Jenkins job to run
+			// and create the new imagestream/tag
+			g.By("verifying that the job ran by looking for the resulting imagestream tag")
+			err := wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
```
You might consider using framework.go:WaitForAnImageStreamTag to keep things DRY. Until #11570 merges, it only waits 1 minute, but you can bump it to 10 and I will rebase #11570 if you merge first.

jupierce commented on test/extended/image_ecosystem/plugin.go:
```diff
@@ -770,6 +771,65 @@ var _ = g.Describe("[image_ecosystem][jenkins][Slow] openshift pipeline plugin",
+
+			// Because polling is enabled, a job should start automatically and fail
+			// Wait long enough for nothing to be running
+			time.Sleep(3 * time.Minute)
```
Perhaps a wait.Poll{ getResource(...) } on the current job colors could eliminate the static wait?
http://stackoverflow.com/questions/14843874/from-jenkins-how-do-i-get-a-list-of-the-currently-running-jobs-in-json

csrwng replied:
sounds better, will do

gabemontero commented:
Noting to add to the discussion that has already taken place.  IGTM.

csrwng commented:
- Addressed comments
- refactored tests to a single function since they're essentially the same except for the job being used. 
- Fixed an issue with the TEST_ONLY=1 case, where the artifact directory was not getting set
- Added a function to specify the timeout when waiting on an ImageStreamTag (@jupierce I thought this may be easier to do if for the most part we want a 1 minute timeout), plus less code to change.

csrwng commented:
For tests to pass, this will require this fix from the jenkins-plugin: https://github.com/openshift/jenkins-plugin/pull/100

csrwng commented:
@gabemontero @jupierce the extended tests now pass, is this ready for merge?
@bparees fyi

gabemontero commented:
interesting ... we've seen some permission issues with projects and the service accounts start occurring over the weekend, but they did not occur with your run .... I see there is no merge conflicts, but did you rebase today?  If not perhaps that is why you did not see these new issues with the extended tests.

That curiosity is not a blocker for me though .... IGTM

csrwng commented:
Yes, I did rebase today. I'm only focusing on the new tests introduced here, so maybe that's why those issues didn't show up?

gabemontero commented:
Yeah perhaps it does not show up with just your test.

bparees commented:
@gabemontero @csrwng 
rebases don't matter, the bot always tests by merging your changes into master before running the test.

(rebases *do* matter for our image repos, because we use the PR tester for those repos and it's less sophisticated.  but for stuff that's managed by the bot, you should never need to rebase to get the bot to test w/ the latest master+your changes)

bparees commented:
[merge]
